services/QuillLMS/db/migrate/20251023174711_convert_examples_to_sti.evidence.rb

```diff
@@ -0,0 +1,180 @@
+# frozen_string_literal: true
+
+# This migration comes from evidence (originally 20251022141000)
+class ConvertExamplesToSti < ActiveRecord::Migration[7.1]
+  # Define models locally to avoid dependency on actual models that may change
+  class Example < ApplicationRecord
+    self.table_name = 'evidence_research_gen_ai_examples'
+    self.inheritance_column = nil # Disable STI for migration to skip validation
+  end
+
+  class PromptExample < ApplicationRecord
+    self.table_name = 'evidence_research_gen_ai_prompt_examples'
+  end
+
+  class TestExample < ApplicationRecord
+    self.table_name = 'evidence_research_gen_ai_test_examples'
+  end
+
+  class DraftExample < ApplicationRecord
+    self.table_name = 'evidence_research_gen_ai_draft_examples'
+  end
+
+  class LlmPromptPromptExample < ApplicationRecord
+    self.table_name = 'evidence_research_gen_ai_llm_prompt_prompt_examples'
+  end
+
+  class LlmExample < ApplicationRecord
+    self.table_name = 'evidence_research_gen_ai_llm_examples'
+  end
+
+  def up
+    create_table :evidence_research_gen_ai_examples do |t|
+      t.text :automl_label
+      t.text :automl_primary_feedback
+      t.text :automl_secondary_feedback
+      t.string :curriculum_assigned_status, null: false
+      t.string :curriculum_label
+      t.text :curriculum_proposed_feedback
+      t.text :highlight
+      t.text :notes
+      t.text :student_response, null: false
+      t.string :type, null: false
+      t.integer :cluster_id
+      t.integer :dataset_id, null: false
+      t.timestamps
+    end
+
+    prompt_example_id_mapping = {}
+    PromptExample.find_each do |prompt_example|
+      Example.create!(
+        prompt_example.attributes.except('id').merge(
+          id: prompt_example.id,
+          type: 'Evidence::Research::GenAi::PromptExample'
+        )
+      )
+      prompt_example_id_mapping[prompt_example.id] = prompt_example.id
+    end
+
+    Rails.logger.debug "Migrated #{prompt_example_id_mapping.size} PromptExamples (IDs unchanged)"
+
+    max_id = Example.maximum(:id) || 0
+
+    test_example_id_mapping = {}
+    TestExample.find_each do |test_example|
+      new_id = test_example.id + max_id
+      Example.create!(
```

Comment:
Just noting that this job is not idempotent. If one of these `create!` errors raises, you'd have to do a bunch of data clean, then restart it (otherwise, it would create duplicates for the ones that were already created, and you'd lose the mapping).

Reply:
Excellent point